src/ott/solvers/nn/models.py

```diff
@@ -0,0 +1,352 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Neural potential models."""
+
+import abc
+from typing import Callable, Optional, Sequence, Tuple, Union
+
+import flax.linen as nn
+import jax
+import jax.numpy as jnp
+import optax
+from flax import struct
+from flax.core import frozen_dict
+from flax.training import train_state
+from jax.nn import initializers
+
+from ott.math import matrix_square_root
+from ott.solvers.nn import layers
+
+__all__ = ["NeuralTrainState", "ModelBase", "ICNN", "MLP"]
+
+PotentialValueFn_t = Callable[[jnp.ndarray], jnp.ndarray]
+PotentialGradientFn_t = Callable[[jnp.ndarray], jnp.ndarray]
+
+
+class NeuralTrainState(train_state.TrainState):
+  """Adds information about the model's value and gradient to the state."""
```

Comment:
Currently, there's no documentation for this, see [here](https://ott-jax--219.org.readthedocs.build/en/219/solvers/_autosummary/ott.solvers.nn.models.NeuralTrainState.potential_value_fn.html#ott.solvers.nn.models.NeuralTrainState.potential_value_fn). Could you please add, e.g., 1 line for the docstring (maybe take from `ModelBase.potential_value_fn`).

Reply:
Added an "Args" section, although that page is still not showing anything. Using "Args" is at least consistent with [flax's docs](https://github.com/google/flax/blob/main/flax/training/train_state.py#L42) for `train_state`.